pkg/server/registry/apigroups/acorn/secrets/storage.go

```diff
@@ -5,18 +5,26 @@ import (
 	"github.com/acorn-io/acorn/pkg/publicname"
 	"github.com/acorn-io/acorn/pkg/tables"
 	"github.com/acorn-io/mink/pkg/stores"
+	"github.com/acorn-io/mink/pkg/strategy"
 	"github.com/acorn-io/mink/pkg/strategy/remote"
 	"github.com/acorn-io/mink/pkg/strategy/translation"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apiserver/pkg/registry/rest"
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func NewStorage(c kclient.WithWatch) rest.Storage {
+type strategyMiddleware func(strategy strategy.CompleteStrategy) strategy.CompleteStrategy
```

Comment:
This type is defined in several different places. Can we move this to be defined in one central place?

Reply:
Yep, I made a `middleware` package for this now. What do you think?